app/models/edu_program.rb

```diff
@@ -21,4 +21,5 @@ class EduProgram < ActiveRecord::Base
   }.freeze
 
   validates :facility_code, presence: true
+  validates_with EduProgramValidator, on: :load_csv
```

Comment:
Why `on: :load_csv`?  Shouldn't this validation be relevant all the time?

Reply:
Without this validations are being run twice in `lib/csv_helper/loader.rb` which was resulting in "missing facility code from weams" being displayed twice per relevant row on upload

The "facility code + description/program name" validations were only failing after all rows were inserted into the table since it is dependent on table data

Reply:
Since I removed the `:validate false` from the import, validations without a context will be run on import and failing rows will end up the `results.failed_instances` array 

This is used at https://github.com/department-of-veterans-affairs/gibct-data-service/blob/5ce4a203d35accf3ae2dda413ee639a5b45303ae/lib/csv_helper/loader.rb#L66

Reply:
The application is setup in a way that only time any validations are run on csv model objects is during csv upload

Reply:
tl;dr: We don't want this validation to prevent records from being persisted to the database. We are providing a validation context so we can manually run the validation post-import to generate warnings for the end user.